lightning-invoice/src/payment.rs

```diff
@@ -698,11 +774,114 @@ mod tests {
 		let scorer = RefCell::new(TestScorer::new());
 		let logger = TestLogger::new();
 		let invoice_payer =
-			InvoicePayer::new(&payer, router, &scorer, &logger, event_handler, RetryAttempts(2));
+			InvoicePayer::new(&payer, router, &scorer, &logger, event_handler, Retry::Attempts(2));
 
 		assert!(invoice_payer.pay_invoice(&invoice).is_ok());
 	}
 
+	#[test]
+	fn fails_paying_invoice_after_max_retry_counts() {
```

Comment:
Is this essentially the same test as `fails_paying_invoice_after_max_retries`?

Reply:
yes previously I change the name to `fails_paying_invoice_after_max_retry_counts` since now we have `fails_paying_invoice_after_max_retry_timeout`. But sorry I probably forgot to remove this after rebase